### ISSUE TITLE
Version-aware `reportShadowedImports`

### ISSUE THREAD

**Issue by ujan-r** [microsoft/pyright#7981]
**Is your feature request related to a problem? Please describe.**

With `reportShadowedImports` enabled, Pyright gives me the following diagnostic when checking a file called `parser.py`:
```
"/PATH/TO/parser.py" is overriding the stdlib module "parser" (reportShadowedImports)
```

However, `parser` was removed in Python 3.10, and I'm using 3.12:
```js
// console.log(execEnv.pythonVersion);
PythonVersion {
  _major: 3,
  _minor: 12,
  _micro: 0,
  _releaseLevel: 'final',
  _serial: 0
}
```

**Describe the solution you’d like**

Only emit a diagnostic if the "shadowed" module actually exists in the current Python version. This would make `reportShadowedImports` much more useful. And since we already store version info for stdlib modules in typeshed, it should be an easy fix!

**Comment by erictraut** [comment]
This feature was added by the pylance team and isn't really a type checking feature. I'll leave it to them to decide whether they want to implement this modification.

Could someone from the pylance team please transfer this issue and triage it as you see fit? Thanks!

cc @rchiodo, @debonte

**Comment by erictraut** [comment]
@ujan-r provided [an implementation](https://github.com/microsoft/pyright/pull/7982), and it looks pretty straightforward. 

Pylance team, please review and if you agree this is aligned with the intent of the feature and the implementation is good, go ahead and merge.

**Comment by debonte** [comment]
Reopening until this ships in a release

**Comment by erictraut** [comment]
This is included in pyright 1.1.365.
